pkg/fs/ext4/ext4.go

```diff
@@ -165,69 +166,37 @@ type EXT4SuperBlock struct {
 	SChecksum         int32      /* crc32c(superblock) */
 }
 
-func (e EXT4SuperBlock) Is() bool {
-	return e.MagicNum == EXT4MagicNum
-}
-
-func (e EXT4SuperBlock) HasExtended() bool {
-	return e.RevLevel >= RevlevelDynamicRev
-}
-
-func (ext4 *EXT4) UUID() (string, error) {
-
-	getUUID := func(ext4uuid [16]uint8) (string, error) {
-		uid, err := uuid.Parse(hex.EncodeToString(ext4uuid[:]))
-		if err != nil {
-			return "", err
-		}
-		return uid.String(), nil
+func (sb *SuperBlock) ID() string {
+	if sb.RevLevel == dynamicRev {
+		return UUID2String(sb.SUuid)
 	}
 
-	if ext4.SuperBlock.HasExtended() {
-		return getUUID(ext4.SuperBlock.SUuid)
-	}
-
-	return strconv.Itoa(int(ext4.SuperBlock.DefaultReserveUID)) + strconv.Itoa(int(ext4.SuperBlock.DefaultReserveGID)), nil
-}
-
-func (ext4 *EXT4) Type() string {
-	return FSTypeEXT4
+	return UUID2String([16]byte{})
```

Comment:
Earlier, if it is not a dynamic revisioned block (Does not have FSUUID), we take `DefaultReserveUID+DefaultReserveGID` as ID.

If it is unique, can we use that as an ID() for dynamic revisioned ext4 drives?

Reply:
`fs-revision-level` i.e. `RevLevel` is set to zero means to be backward compatible with linux kernel <= 1.2. In this case, using of default reserved GID/UID does not help to get uniqueness of different devices. Returning zero and generating new UUID at usage is more appropriate.